engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb

```diff
@@ -2,61 +2,106 @@ module Onboarding
   module Api
     module V1
       class InvitationsController < ApplicationController
+        before_action :authenticate, only: [:bulk]
+        before_action :load_group, only: [:bulk]
+        before_action :load_invitation, only: [:show, :accept]
 
-        before_action :authenticate
-        before_action :load_group, only: [:create]
-        before_action :load_invited_user, only: [:create]
+        # Creates invitations in bulk
+        #
+        # POST /api/v1/invitations/bulk
+        #
+        def bulk
+          authorize @group
+
+          emails = bulk_params[:emails] || ''
+
+          if emails.empty?
+            return render(
+              status: :bad_request,
+              json: {errors: {emails: t('onboarding.invitation.bulk.errors.empty') }}
+            )
+          end
+
+          emails = emails.split(',')
+          valids = valid_emails(emails)
+
+          if valids.any?
+            # We pick first 100 valid emails
+            # We don't want to allow infinite invitations.
+            valid_emails = valids.first(100)
+            InvitationService.new.bulk_invite!(@group, current_user, valid_emails)
+
+            render nothing: true, status: :accepted
+          else
+            render(
+              status: :bad_request,
+              json: {errors: {emails: t('onboarding.invitation.bulk.errors.invalid') }}
+            )
+          end
+        end
+
+        #
+        # GET /invitations/:token
+        #
+        def show
+          if current_user
+            render status: :not_found, nothing: true
+          else
+            render status: :ok, json: { email: @invitation.email }
+          end
+        end
```

Comment:
I want to know if invitation esists. Other way I will show a Comple signup Form for whatever ramdom token